client/app/components/product-card/index.jsx

```diff
@@ -51,26 +42,69 @@ const ProductLink = style(Link)`
   outline: 0;
   text-decoration: none;`;
 
-const Card = ({ product, quickview, open, close}) => (
-  <OuterShell>
-    <InnerShell>
-      <CardBox>
-        <ProductLink href={product._id}>
-          <Image product={product} />
-          <Name product={product} />
-          <Brand product={product} />
-          <Price product={product} />
-          <Reviews product={product} />
-        </ProductLink>
-      </CardBox>
-      <Description product={product} hide={!quickview} close={close} />
-    </InnerShell>
-    <QuickViewButton
-      name="Quickview"
-      handleClick={() => open(product._id)}
-      removed={quickview}
-    />
-  </OuterShell>
-);
+const Card = (props) => {
+  const {
+    product,
+    quickview,
+    open,
+    close,
+  } = props;
+  return (
+    <OuterContainer>
+      <InnerContainer>
+        <CardBox>
+          <ProductLink href={product.url}>
+            <Image imageURL={product.image_url} name={product.name} />
+            <Name name={product.name} />
+            <Brand brand={product.brand} />
+            <Price price={product.price} />
+            <Reviews
+              score={product.review_score}
+              count={product.review_count}
+            />
+          </ProductLink>
+        </CardBox>
+        <Description
+          url={product.url}
+          description={product.description}
+          hide={!quickview}
+          close={close}
+          fullDetailsLinkTitle="See Full Details"
+        />
+      </InnerContainer>
+      <QuickViewButton
+        title="Quickview"
+        click={() => open(product.id)}
+        removed={quickview}
+      />
+    </OuterContainer>
+  );
+};
+
+Card.propTypes = {
+  product: PropTypes.shape({
+    _id: PropTypes.string.isRequired,
+    id: PropTypes.string.isRequired,
+    url: PropTypes.string.isRequired,
+    name: PropTypes.string.isRequired,
+    description: PropTypes.string,
+    image_url: PropTypes.shape({
+      small: PropTypes.string.isRequired,
+      medium: PropTypes.string.isRequired,
+      large: PropTypes.string.isRequired,
+    }),
+    brand: PropTypes.string.isRequired,
+    price: PropTypes.string.isRequired,
+    review_score: PropTypes.number.isRequired,
+    review_count: PropTypes.number.isRequired,
+  }),
+  quickview: PropTypes.bool.isRequired,
+  open: PropTypes.func.isRequired,
+  close: PropTypes.func.isRequired,
+};
+
+Card.defaultProps = {
+  product: PropTypes.object,
+};
```

Comment:
I think defaultProps should have an actual value - `( { hello: 'word' } )` for example  - not a propType.

Reply:
I will look into that. Thanks for the advice.

Reply:
I fixed this. I had to require composites in type checks. If fine tunning necessitates, I will gradually provide defaults.